src/main/java/picocli/CommandLine.java

```diff
@@ -3650,6 +3677,15 @@ public CommandSpec aliases(String... aliases) {
                 return this;
             }
 
+            /** Returns the default value provider for this command.
+             * @return the default value provider or {@code null}*/
+            public IDefaultValueProvider defaultValueProvider() { return defaultValueProvider; }
+
+            /** Sets default value provider for this command.
+             * @param defaultValueProvider the default value provider to use, or {@code null}.
+             * @return this CommandSpec for method chaining */
+            public CommandSpec defaultValueProvider(IDefaultValueProvider  defaultValueProvider) { this.defaultValueProvider = defaultValueProvider; return this; }
+
```

Comment:
I am wondering whether it would be a good idea to also add programmatic API for setting/getting the `IDefaultProvider` on the `CommandLine` facade class. 

If we think this is a good idea, it should work like the other `CommandLine.getX` and `setX` methods: apply the new X value to all subcommands that are registered at the time of the call. (See for example setStopAtPositional, setToggleBooleanFlags etc).